flexget/plugins/api_tmdb.py

```diff
@@ -269,6 +279,132 @@ def lookup(title=None, year=None, tmdb_id=None, imdb_id=None, smart_match=None,
         return movie
 
 
+# Oauth account authentication
+class TmdbUserAuth(Base):
+    __tablename__ = 'tmdb_user_auth'
+
+    account = Column(Unicode, primary_key=True)
+    session_id = Column(Unicode)
+    created = Column(DateTime)
+
+    def __init__(self, account, session_id, created=datetime.utcnow()):
+        self.account = account
+        self.session_id = session_id
+        self.created = created
+
+
+def authenticate(account):
+    """Authenticates a session with TMDB, and grabs any IDs needed for getting/modifying list."""
+    try:
+        r = tmdb_request('authentication/token/new')
+        url = r.headers['Authentication-Callback']
+        body = r.json()
+        request_token = body['request_token']
+        expires_at = body['expires_at']
+
+        console('Please visit {0} and authorize Flexget. Your token expires {1}.'.format(url, expires_at))
+        input('Press `Enter` when the token has been authorized...')
```

Comment:
This isn't ideal, a loop would be better, like in trakt:
https://github.com/Flexget/Flexget/blob/develop/flexget/plugins/api_trakt.py#L83